bundles/org.dataflowanalysis.analysis.dfd/src/org/dataflowanalysis/analysis/dfd/core/DFDFlowGraphCollection.java

```diff
@@ -57,6 +68,19 @@ public List<? extends AbstractTransposeFlowGraph> findTransposeFlowGraphs() {
             logger.error("Cannot find transpose flow graphs for non-dfd resource provider");
             throw new IllegalArgumentException();
         }
-        return new DFDTransposeFlowGraphFinder(dfdResourceProvider).findTransposeFlowGraphs();
+
+        if (transposeFlowGraphFinderClass.equals(DFDTransposeFlowGraphFinder.class))
+            this.transposeFlowGraphFinder = new DFDTransposeFlowGraphFinder(dfdResourceProvider);
+        else
+            this.transposeFlowGraphFinder = new DFDCyclicTransposeFlowGraphFinder(dfdResourceProvider);
+
+        return transposeFlowGraphFinder.findTransposeFlowGraphs();
+    }
+
+    public boolean wasCyclic() {
+        if (transposeFlowGraphFinder instanceof DFDCyclicTransposeFlowGraphFinder cyclicTransposeFlowGraphFinder) {
+            return cyclicTransposeFlowGraphFinder.hasCycles();
+        }
+        return false;
     }
```

Comment:
Doc: Missing documentation for public method.
Furthermore, add warning here for calling on flow graph collection without cyclic transpose flow graph finder